src/terminal/input/terminalInput.cpp

```diff
@@ -498,16 +498,16 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent)
         if (ch == UNICODE_NULL)
         {
             // For Alt+Ctrl+Key messages GetCharData() returns 0.
+            // The values of the ASCII characters and virtual key codes
+            // of <Space>, A-Z (as used below) are numerically identical.
             // -> Get the char from the virtual key.
-            ch = LOWORD(MapVirtualKeyW(keyEvent.GetVirtualKeyCode(), MAPVK_VK_TO_CHAR));
+            ch = keyEvent.GetVirtualKeyCode();
         }
-        if (ch == UNICODE_SPACE)
-        {
-            // Ctrl+@ and Ctrl+Space are supposed to send null bytes.
-            // -> Change Ctrl+Space to Ctrl+@ for compatibility reasons.
-            ch = 0x40;
-        }
-        if (ch >= 0x40 && ch < 0x7F)
+        // Alt+Ctrl acts as a substitute for AltGr on Windows.
+        // For instance using a German keyboard both AltGr+< and Alt+Ctrl+< produce a | (pipe) character.
+        // The below condition primitively ensures that we allow all common Alt+Ctrl combinations
+        // while preserving most of the functionality of Alt+Ctrl as a substitute for AltGr.
+        if (ch == UNICODE_SPACE || (ch > 0x40 && ch <= 0x5A))

```

Comment:
Oh, I see: this change is safe because `0x20 & 0b11111` is 0 (we didn't need to set it to 0x40 just to strip off the top few bits). Clever.

Reply:
@DHowett-MSFT Did you ever see this article? https://garbagecollected.org/2017/01/31/four-column-ascii/
You probably know everything about this already, but for me it was pretty eye opening regardless...
It quickly shows you _why_ `^␣` is actually the same as `^@` and why `^[` is the same as the ESC character. It's a much better representation of ASCII as the square one you can find everywhere else (including Wikipedia). 🙂

Reply:
I actually used to have a giant version of [this table](http://www.asciitable.com/) printed in my office, which has been immensely helpful